canonical-web-and-design/maas-ui
# Add noderesult slice

## Done

- Add noderesult slice (note, scriptResults slice will be removed in a followup PR)
- Add a `nocache` meta param, which prevents allows `list` methods to be optionally called repeatedly

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- visit the machine details testing tab
- ensure nodeResults for that machine appear in redux state
- visit the machine details testing tab for a different machine
- ensure those nodeResult appear, and that the previous machine's are still in state

## Fixes

Fixes: canonical-web-and-design/maas-ui#1901.

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.
